search_service/api/filter.py

```diff
@@ -0,0 +1,95 @@
+from http import HTTPStatus
+from typing import Any, Dict, Iterable  # noqa: F401
+
+from flask_restful import Resource, reqparse
+from flasgger import swag_from
+from marshmallow_annotations.ext.attrs import AttrsSchema
+
+from search_service.api.table import TABLE_INDEX
+from search_service.models.table import SearchTableResultSchema
+from search_service.models.dashboard import SearchDashboardResultSchema
+from search_service.proxy import get_proxy_client
+
+
+class BaseFilterAPI(Resource):
+    """
+    Base Filter API for search filtering
+
+    This API should be generic enough to support every search filter use case.
+    """
+
+    def __init__(self, *, schema: AttrsSchema) -> None:
+        self.proxy = get_proxy_client()
+        self.schema = schema
+        self.parser = reqparse.RequestParser(bundle_errors=True)
+
+        self.parser.add_argument('index', required=False, default=TABLE_INDEX, type=str)
```

Comment:
How is it that this will know which index to use, and where does that information come from? Is the FE API call supposed to include `index` now? If so it would be nice to have the values like `TABLE_INDEX = 'table_search_index'` defined in `amundsen_common` too.

Reply:
make sense, we could do it in a follow pr to clean this up.